packages/core/src/lib/cls.service.ts

```diff
@@ -125,10 +125,10 @@ export class ClsService<S extends ClsStore = ClsStore> {
         let options: ClsContextOptions;
         let callback: () => any;
         if (typeof optionsOrCallback === 'object') {
-            options = optionsOrCallback;
+            options = { ...new ClsContextOptions(), ...optionsOrCallback };
             callback = maybeCallback;
         } else {
-            options = { ...new ClsContextOptions(), ...optionsOrCallback };
+            options = new ClsContextOptions();
```

Comment:
(optional) This change is not really needed, but I'm guessing this code was introduced by mistake, and that the behavior was inverted.